TripalCultivate/TripalCultivate
# Update the automated testing to check new versions of Drupal

**Issue #33**

## Motivation

We need to support the more recent versions of Drupal 🤷‍♀️ 

## What does this PR do?


- [x] Update GitHub workflows to include Drupal 10.5, 11.0, + 11.1 and remove 10.2.
- [x] Split the current dockerfile into two parts: 1) install of Tripal, Chado, Theme; 2) install of TripalCultivate Base.
- [x] Update devcontainer to run on Drupal 11.1 and to use the new dockerfile setup.

## Testing

### Automated Testing


This PR does not add any additional automated tests. Instead it simply expands the versions these tests are ran on. 

### Manual Testing


1. Checkout this branch locally, open it in VSCode and "Reopen in Container".
2. Confirm that devcontainer builds properly (i.e. no screaming error 😉) and open the webpage in the browser to confirm versions. Note it is now building Drupal 11.1
<img width="656" alt="Screenshot 2024-11-21 at 10 05 48 AM" src="https://github.com/user-attachments/assets/01abe7a3-d2e6-46c1-b8e4-88c8961ec260">

3. Also check the README on this branch by going to the following link and confirm the testing grid renders properly with it's badges. It should look like this:
https://github.com/TripalCultivate/TripalCultivate/tree/g5.33-supportD11?tab=readme-ov-file#automated-testing
<img width="1125" alt="Screenshot 2024-11-21 at 10 07 43 AM" src="https://github.com/user-attachments/assets/2b2bbc67-c2c9-407c-b373-9150716c0d0e">
